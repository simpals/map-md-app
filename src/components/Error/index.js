import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import errorImage from '../../static/images/error-page-holder.svg';

const Container = styled.div`
  text-align: center;
  padding: 140px 0 70px;
  & > * {
    font-family: SF Pro Display;
  }
`;

const ErrorImage = styled.img`
  display: inline-block;
  margin-bottom: 25px;
`;

const ErrorText = styled.p`
  color: rgba(0, 0, 0, 0.4);
  margin: 10px auto;
  font-size: 13px;
  line-height: 15px;
  text-align: center;
  max-width: 450px;
`;

const Error = ({ error }) => (
  <Container>
    <ErrorImage src={errorImage} alt='error image' />
    <ErrorText>Something went wrong</ErrorText>
    <ErrorText>{`Error message: ${error}.`}</ErrorText>
  </Container>
);

Error.defaultProps = {
  error: null,
};

Error.propTypes = {
  error: PropTypes.any,
};

export default Error;
