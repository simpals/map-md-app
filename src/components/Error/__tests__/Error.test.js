import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Error from '..';

const testError = Error('TestMessage', 'TestFile', 'line 1:1');
let component;

beforeEach(() => {
  component = mount(<Error />);
});

describe('test Error component', () => {
  it('renders without crashing', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('has default props', () => {
    expect(component.props().error).toEqual(null);
  });

  it('has props', () => {
    component.setProps({
      error: testError
    });
    expect(component.props().error).toEqual(testError);
  });
});
