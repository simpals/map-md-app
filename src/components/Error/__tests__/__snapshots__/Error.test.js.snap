// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Error component renders without crashing 1`] = `
<Error
  error={null}
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "cebmy",
            "rules": Array [
              "
  text-align: center;
  padding: 140px 0 70px;
  & > * {
    font-family: SF Pro Display;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa cebmy"
      >
        <styled.img
          alt="error image"
          src="test-file-stub"
        >
          <StyledComponent
            alt="error image"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "hPnSzP",
                  "rules": Array [
                    "
  display: inline-block;
  margin-bottom: 25px;
",
                  ],
                },
                "displayName": "styled.img",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "img",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            src="test-file-stub"
          >
            <img
              alt="error image"
              className="sc-bwzfXH hPnSzP"
              src="test-file-stub"
            />
          </StyledComponent>
        </styled.img>
        <styled.p>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": true,
                  "lastClassName": "jdCEQr",
                  "rules": Array [
                    "
  color: rgba(0, 0, 0, 0.4);
  margin: 10px auto;
  font-size: 13px;
  line-height: 15px;
  text-align: center;
  max-width: 450px;
",
                  ],
                },
                "displayName": "styled.p",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "p",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <p
              className="sc-htpNat jdCEQr"
            >
              Something went wrong
            </p>
          </StyledComponent>
        </styled.p>
        <styled.p>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": true,
                  "lastClassName": "jdCEQr",
                  "rules": Array [
                    "
  color: rgba(0, 0, 0, 0.4);
  margin: 10px auto;
  font-size: 13px;
  line-height: 15px;
  text-align: center;
  max-width: 450px;
",
                  ],
                },
                "displayName": "styled.p",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "p",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <p
              className="sc-htpNat jdCEQr"
            >
              Error message: null.
            </p>
          </StyledComponent>
        </styled.p>
      </div>
    </StyledComponent>
  </styled.div>
</Error>
`;
